metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/MonitoredExecutorService.java

```diff
@@ -0,0 +1,89 @@
+package com.ringcentral.platform.metrics.instrument.executors;
+
+import com.ringcentral.platform.metrics.MetricKey;
+import com.ringcentral.platform.metrics.MetricRegistry;
+import com.ringcentral.platform.metrics.labels.LabelValues;
+import com.ringcentral.platform.metrics.names.MetricName;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.function.Function;
+
+import static com.ringcentral.platform.metrics.names.MetricName.name;
+
+/**
+ * An {@link ExecutorService} wrapper that provides the following metrics:
+ * <ul>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for submitted tasks, with the name {@code metricKeyProvider.apply(name("submitted"))}</li>
```

Comment:
Suggest to move it to `AbstractMonitoredExecutorService` and leave here a link to it.

Reply:
Agreed.

Reply:
Unfortunately I'm not ready to agree with statement about inconvenience, but I don't insist on this change.